rust/trace-dump/src/chrome_trace.rs

```diff
@@ -31,172 +23,10 @@ pub enum Error {
-    // Split it back out into a vector.
-    let categories = trace_entry.cat.split(",").map(
-        |s| s.to_string()).collect::<Vec<String>>();
-
-    let (sample_id, payload ) = trace_entry.args.map_or((default_sample_id, None), |args| {
-        (args.xi_sample_id.unwrap_or(default_sample_id), args.payload)
-    });
-
-    let ph = trace_entry.ph.as_ref();
-
-    let mut converted = match ph {
-        "i" => Ok(Sample::new_instant(trace_entry.name, categories, payload)),
-        "B" => Ok(Sample::new(trace_entry.name, categories, payload)),
-        _ => Err(Error::DecodingFormat(
-                format!("Entry has unexpected ph value {}",
-                        ph)))
-    }?;
-
-    converted.sample_id = sample_id;
-    converted.pid = trace_entry.pid;
-    converted.tid = trace_entry.tid;
-    converted.start_ns = to_ns(trace_entry.ts_us);
-    Ok(converted)
-}
-
-impl XiTryFrom<Vec<ChromeTraceArrayEntry>> for Vec<Sample> {
-    type Error = Error;
-
-    /// Converting Chrome trace event back into a Sample is a bit more work
-    /// because a Sample can represent a duration so we have to merge Chrome
-    /// trace events that indicate start/end.
-    fn try_from(trace_entries: Vec<ChromeTraceArrayEntry>) -> Result<Self, Error> {
-        let mut result = Vec::with_capacity(trace_entries.len());
-        for trace_entry in trace_entries {
-            if trace_entry.ph == "E" {
-                // Got an end of a duration measure so look back into the
-                // samples we've already converted to populate the end
-                // timestamp.
-                let mut found = false;
-                for mut existing_sample in result.iter_mut().rev() {
-                    if is_begin_sample(&existing_sample, trace_entry.pid, trace_entry.tid, &trace_entry.name) {
-                        existing_sample.set_end_ns(to_ns(trace_entry.ts_us));
-                        found = true;
-                        break;
-                    }
-                }
-                if !found {
-                    return Err(Error::DecodingFormat(
-                        format!("Entry {:?} found but no preceding start sample exists",
-                                trace_entry)));
-                }
-            } else {
-                let default_sample_id = result.len();
-                result.push(try_from(trace_entry, default_sample_id)?);
-            }
-        }
-        Ok(result)
-    }
-}
-
-impl XiTryFrom<ChromeTraceArrayEntries> for Vec<Sample> {
-    type Error = Error;
-
-    /// Converting Chrome trace event back into a Sample is a bit more work
-    /// because a Sample can represent a duration so we have to merge Chrome
-    /// trace events that indicate start/end.
-    fn try_from(entries: ChromeTraceArrayEntries) -> Result<Self, Error> {
-        match entries {
-            ChromeTraceArrayEntries::Array(samples_array) => Vec::try_from(samples_array),
-        }
-    }
+    Array(Vec<Sample>)
```

Comment:
Is this enum necessary? If this is just about serialization we can always just put a `Vec<Sample>` on the wire, we don't need a wrapper type.

Reply:
Yeah, I couldn't figure out how to get the code to compile without this dummy enum.  If you have suggestions, happy to implement it.

Reply:
No worries, especially if we're going to get rid of this crate at some point.